src/software/thunderscope/constants.py

```diff
@@ -300,6 +302,83 @@ class TrailValues:
     DEFAULT_TRAIL_SAMPLING_RATE = 0
 
 
+@dataclass
+class DeviceKeyEvent:
+    """This dataclass holds the code for a "EV_KEY" input event"""
+
+    event_code: int
+
+
+@dataclass
+class DeviceAbsEvent:
+    """This dataclass holds the code and max value for an "EV_ABS" input event"""
+
+    event_code: int
+    max_value: float
+
+
+@dataclass
+class DeviceConfig:
+    move_x: DeviceAbsEvent
+    move_y: DeviceAbsEvent
+    move_rot: DeviceAbsEvent
+    kick: DeviceKeyEvent
+    chip: DeviceKeyEvent
+    chicker_power: DeviceAbsEvent
+    dribbler_speed: DeviceAbsEvent
+    dribbler_enable: DeviceAbsEvent
+
+
+class DiagnosticsConstants:
+    XboxConfig = DeviceConfig(
+        # Name: "ABS_X"
+        # Canonical: Left joystick X-axis
+        move_x=DeviceAbsEvent(event_code=0, max_value=32767.0),
+        # Name: "ABS_Y"
+        # Canonical: Left joystick Y-axis
+        move_y=DeviceAbsEvent(event_code=1, max_value=32767.0),
+        # Name: "ABS_RX"
+        # Canonical: Right joystick X-axis
+        move_rot=DeviceAbsEvent(event_code=3, max_value=32767.0),
+        # Name: "BTN_A"
+        # Canonical: "A" Button
+        kick=DeviceKeyEvent(event_code=304),
+        # Name: "BTN_Y"
+        # Canonical: "Y" Button
+        chip=DeviceKeyEvent(event_code=308),
+        # Name: "ABS_HAT0X
+        # Canonical: D-pad X-axis
+        chicker_power=DeviceAbsEvent(event_code=16, max_value=1.0),
+        # Name: "ABS_HAT0Y"
+        # Canonical: D-pad Y-axis
+        dribbler_speed=DeviceAbsEvent(event_code=17, max_value=1.0),
+        # Name: "ABS_RZ"
+        # Canonical: Right trigger
+        dribbler_enable=DeviceAbsEvent(event_code=5, max_value=1023.0),
+    )
+
+    HANDHELD_DEVICE_NAME_CONFIG_MAP = {
```

Comment:
Now that I think about it, does it make sense to even make sense to have this `CONFIG_MAP`. The only configruation we currently have is `XboxConfig`. 

We could literally just have a list and call it `supported_deviced` and initialize this `XboxConfig`  if the device we are looking at is in this list. 

Are we even going to support different controller in the future?

Reply:
This is probably a YAGNI moment. But I'll leave the config map as is bc it's already implemented

Reply:
I imagine at some point we might want to add a generic controller or a play station controller or something, so may as well